ros2_batch_job/__main__.py

```diff
@@ -371,7 +371,8 @@ def build_and_test(args, job):
             'import pytest;'
             'import sys;'
             'sys.exit(StrictVersion(pytest.__version__) >= StrictVersion("6.0.0"))'
-            "'"])
+            "'"],
+            exit_on_error=False)
```

Comment:
Using the return value for both the version comparison result as well as potential errors looks fragile to me. I would suggest to refactor this logic to:
* either print the version comparison result (and interpret the output) and ensure that the command has a return code of zero
* or just print `pytest.__version__` and perform the comparison in the calling Python code rather than within the invoked command.